provider/openstack/cinder.go

```diff
@@ -262,7 +262,7 @@ func (s *cinderVolumeSource) createVolume(arg storage.VolumeParams) (*storage.Vo
 		if err := s.storageAdapter.DeleteVolume(volumeId); err != nil {
 			logger.Warningf("destroying volume %s: %s", volumeId, err)
 		}
-		return nil, errors.Errorf("waiting for volume to be provisioned: %s", err)
+		return nil, HandleCredentialError(errors.Errorf("waiting for volume to be provisioned: %s", err), ctx)
```

Comment:
As a drive by, change to error.Annotate(err, "waiting for volume to be provisioned")

here and anywhere else touched by the changes